crux_core/src/typegen.rs

```diff
@@ -355,6 +355,11 @@ impl TypeGen {
 
         fs::create_dir_all(&path)?;
 
+        let package_path = package_name.replace('.', "/");
+
+        // remove any existing generated shared types, this ensures that we remove no longer used types
+        fs::remove_dir_all(path.as_ref().to_path_buf().join(&package_path)).unwrap_or(());
```

Comment:
I think we can remove the `to_path_buf()` call, as `join()` is on `&path`. This will avoid an unneccesary coercion.

Reply:
I removed the unnecessary coercion. Thank you for the bringing that to my attention.